projects/packages/jetpack-mu-wpcom/src/features/calypso-locale-sync/sync-wp-admin-to-calypso.php

```diff
@@ -63,7 +76,12 @@ function sync_wp_admin_locale_to_calypso( $meta, $user, $update ) {
  */
 function sync_wp_admin_site_locale_with_site_default_to_calypso( $old_value, $new_value ) {
 	if ( empty( get_user_option( 'locale' ) ) ) {
-		// No user locale means to use the site language (WPLANG) or "en".
+		// Empty user locale means site-default and uses the site language (WPLANG) or "en" when WPLANG is empty.
 		update_calypso_locale( $new_value ? $new_value : 'en' );
 	}
 }
+
+if ( function_exists( 'wpcom_is_nav_redesign_enabled' ) && wpcom_is_nav_redesign_enabled() ) {
```

Comment:
I believe this is only intended for WoA sites, should we bail on Simple sites?

Reply:
Because there is Jetpack connection on Simple sites, the `$is_user_connected` check a few lines above will effectively bail on Simple sites.

But I guess we can add a more explicit check to clearly indicate this logic is not meant to run on Simple sites.

Reply:
https://github.com/Automattic/jetpack/pull/37750/commits/25682ef1fcff140eeb1c84ade1689298fca8eb33